.github/workflows/tests.yml

```
@@ -105,6 +112,9 @@ jobs:
         run: |
           hash -r
           env | sort
+      - name: Create private key for network testing
+        if: ${{ matrix.test-case == 'test-local' }}
+        run: ${{ matrix.test-option }} make create-private-key
```

Is this for non-remote network unit tests? Remote network tests does not need it as well?
Are non-remote network tests expected to run along other unit tests? What if network mode is disabled for that instance? They will automatically fail?

---

> Is this for non-remote network unit tests?

Yes

> Remote network tests does not need it as well?

It does but in that case it is created automatically when the app starts by setting the `MAGPIE_NETWORK_CREATE_MISSING_PEM_FILE` variable.

> Are non-remote network tests expected to run along other unit tests?

Yes

> What if network mode is disabled for that instance?

For local tests, network mode will be enabled for that specific test case.
For remote tests, the test will be skipped.

See:

https://github.com/Ouranosinc/Magpie/blob/fa4f9e8bc073c15220ad493176f22a03ddc31cf7/tests/utils.py#L743-L760

> They will automatically fail?

No

---

OK. I just want to double-check my understanding of the test suite. 

Since the new test methods are applied with the `@utils.check_network_mode` decorator onto existing `unittest.TestCase`, and that those classes' setup invoke `cls.setup_network_attrs()`, they are all running with *Network Mode* enabled? Shouldn't they fail in this case instead if the network was not configured correctly? That would validate that configuration setup also works, not silently ignored.

Also, doesn't that mean that nothing tests these API calls with *Network Mode* disabled anymore? 

I would expect to have distinct derived unittest classes that evaluate with/without *Network Mode*. This is similar to how the various https://github.com/Ouranosinc/Magpie/blob/master/tests/interfaces.py classes are reused to validate different branches in the code depending on enabled features or not.